src/ion/process.rs

```diff
@@ -1096,13 +1155,18 @@ impl<'a, F: Function> Env<'a, F> {
                     (Some(a), Some(b)) => Some(core::cmp::max(a, b)),
                     _ => None,
                 };
-                match self.try_to_allocate_bundle_to_reg(bundle, preg_idx, scan_limit_cost) {
+                match self.try_to_allocate_bundle_to_reg(
+                    bundle,
+                    preg_idx,
+                    scan_limit_cost,
+                    &mut scratch,
+                ) {
                     AllocRegResult::Allocated(alloc) => {
-                        self.stats.process_bundle_reg_success_any += 1;
+                        self.ctx.output.stats.process_bundle_reg_success_any += 1;
                         trace!(" -> allocated to any {:?}", preg_idx);
-                        self.spillsets[self.bundles[bundle].spillset].reg_hint =
+                        self.ctx.spillsets[self.ctx.bundles[bundle].spillset].reg_hint =
                             alloc.as_reg().unwrap();
-                        return Ok(());
+                        break 'o;
```

Comment:
Can we choose a more descriptive name for this loop label (`'outer` or somesuch) and add a comment here (and below) where we replaced `return Ok(())` to indicate this is a successful allocation ("`// Success, return scratch memory to context and finish`" or similar)?